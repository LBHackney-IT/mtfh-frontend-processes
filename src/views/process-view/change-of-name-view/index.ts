export * from "./change-of-name-view";
export * from "./states";
