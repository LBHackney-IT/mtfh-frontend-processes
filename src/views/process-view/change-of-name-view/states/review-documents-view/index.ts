export * from "./review-documents-view";
