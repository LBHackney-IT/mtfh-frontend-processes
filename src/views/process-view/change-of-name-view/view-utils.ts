import { locale, processes } from "../../../services";

const { reviewDocuments } = locale.views;

const { states } = processes.changeofname;

const { interviewScheduled, interviewRescheduled, hoApprovalFailed, hoApprovalPassed } =
  states;

export const changeOfNameDocuments = [
  reviewDocuments.marriageCertificate,
  reviewDocuments.civilPartnershipCertificate,
  reviewDocuments.decreeAbsolute,
  reviewDocuments.finalOrder,
  reviewDocuments.deedPollDocument,
  reviewDocuments.statuoryDeclaration,
];

export const cancelButtonStates = [
  interviewScheduled.state,
  interviewRescheduled.state,
  hoApprovalFailed.state,
  hoApprovalPassed.state,
];
