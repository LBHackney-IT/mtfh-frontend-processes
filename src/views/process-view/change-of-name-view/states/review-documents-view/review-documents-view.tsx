import { Form, Formik } from "formik";

import {
  ReviewDocumentsFormData,
  reviewDocumentsSchema,
} from "../../../../../schemas/change-of-name/review-documents";
import { locale } from "../../../../../services";
import { Trigger } from "../../../../../services/processes/types";
import { IProcess } from "../../../../../types";
import { ReviewDocumentsAppointmentForm } from "../../../shared/review-documents-appointment-form";
import { changeOfNameDocuments } from "../../view-utils";

import { Process, editProcess } from "@mtfh/common/lib/api/process/v1";
import {
  Button,
  Center,
  Checkbox,
  CheckboxGroup,
  FormGroup,
  Heading,
  InlineField,
  Spinner,
} from "@mtfh/common/lib/components";
import { useErrorCodes } from "@mtfh/common/lib/hooks";

const { views } = locale;
const { reviewDocuments } = views;
interface ReviewDocumentsViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
  setGlobalError: any;
}

export const ReviewDocumentsView = ({
  processConfig,
  process,
  mutate,
  optional,
  setGlobalError,
}: ReviewDocumentsViewProps) => {
  const errorMessages = useErrorCodes();

  if (!errorMessages) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  return (
    <div data-testid="changeofname-ReviewDocuments">
      <ReviewDocumentsAppointmentForm
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        setGlobalError={setGlobalError}
      />

      {!optional?.closeProcessReason && (
        <>
          <div style={{ paddingBottom: 35 }} />

          <Formik<ReviewDocumentsFormData>
            initialValues={{
              seenPhotographicId: false,
              seenSecondId: false,
              atLeastOneDocument: false,
            }}
            validateOnBlur={false}
            validateOnChange={false}
            validationSchema={reviewDocumentsSchema(errorMessages)}
            onSubmit={async (values) => {
              try {
                await editProcess({
                  id: process.id,
                  processTrigger: Trigger.ReviewDocuments,
                  processName: process?.processName,
                  etag: process.etag || "",
                  formData: values,
                  documents: [],
                });
                mutate();
              } catch (e: any) {
                setGlobalError(e.response?.status || 500);
              }
            }}
          >
            {({ values, errors }) => {
              return (
                <Form
                  noValidate
                  id="review-documents-form"
                  className="review-documents-form"
                >
                  <Heading variant="h4">{reviewDocuments.useFormBelow}</Heading>
                  <FormGroup
                    id="review-documents-form-group"
                    label=""
                    error={
                      errors.seenPhotographicId ||
                      errors.seenSecondId ||
                      errors.atLeastOneDocument
                    }
                  >
                    <CheckboxGroup>
                      <InlineField name="seenPhotographicId" type="checkbox">
                        <Checkbox
                          id="seen-photographic-id"
                          hint={reviewDocuments.seenPhotographicIdHint}
                        >
                          {reviewDocuments.seenPhotographicId}
                        </Checkbox>
                      </InlineField>
                      <InlineField name="seenSecondId" type="checkbox">
                        <Checkbox
                          id="seen-second-id"
                          hint={reviewDocuments.seenSecondIdHint}
                        >
                          {reviewDocuments.seenSecondId}
                        </Checkbox>
                      </InlineField>
                      <InlineField name="atLeastOneDocument" type="checkbox">
                        <Checkbox id="confirmation-for-valid-documents">
                          {reviewDocuments.atLeastOneDocument}
                          {changeOfNameDocuments.map((documentName, index) => {
                            return (
                              <>
                                <br />
                                {index + 1}.{documentName}
                              </>
                            );
                          })}
                        </Checkbox>
                      </InlineField>
                    </CheckboxGroup>
                  </FormGroup>
                  <Button
                    type="submit"
                    disabled={
                      !Object.values(values).some((value) => {
                        return value;
                      })
                    }
                    style={{ width: 222 }}
                  >
                    {locale.next}
                  </Button>
                </Form>
              );
            }}
          </Formik>
        </>
      )}
    </div>
  );
};
