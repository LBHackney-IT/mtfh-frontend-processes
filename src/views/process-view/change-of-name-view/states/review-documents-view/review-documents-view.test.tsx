import React from "react";

import {
  getReferenceDataV1,
  mockPersonV1,
  mockProcessV1,
  patchProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { fireEvent, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../../services";
import {
  mockDocumentsRequestedAppointment,
  mockDocumentsRequestedDes,
  mockDocumentsRequestedDesAppointment,
  typeDateTime,
} from "../../../../../test-utils";
import { ChangeOfNameView } from "../../change-of-name-view";
import { ReviewDocumentsView } from "./review-documents-view";

import * as errorMessages from "@mtfh/common/lib/hooks/use-error-codes";

const options = {
  url: "/processes/changeofname/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

describe("review-documents-view", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders ReviewDocuments correctly on state DocumentsRequestedDes", async () => {
    jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
    const { container } = render(
      <ChangeOfNameView
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        optional={{}}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.viewInDes),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.reviewDocuments.checkSupportingDocumentsAppointment),
    ).resolves.toBeInTheDocument();
    await expect(screen.getByText(locale.next)).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it enables book appointment once checkbox is selected", async () => {
    server.use(getReferenceDataV1({}, 200));
    render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
    );
    await expect(screen.queryByText(locale.bookAppointment)).toBeNull();
    const checkbox = screen.getByLabelText(
      locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
    );
    fireEvent.click(checkbox);
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeDisabled();
  });

  test("it enable book appointment button if date is typed in", async () => {
    server.use(getReferenceDataV1({}, 200));
    server.use(patchProcessV1("error", 500));
    render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
    );
    const checkbox = screen.getByLabelText(
      locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
    );
    await userEvent.click(checkbox);
    await typeDateTime(screen, userEvent, "2000");
    await userEvent.click(screen.getByText(locale.bookAppointment));
    await expect(screen.findByText(locale.bookAppointment)).resolves.toBeEnabled();
  });

  test("it displays an error if there's an issue with book appointment", async () => {
    const user = userEvent.setup();
    server.use(patchProcessV1("error", 500));
    server.use(getReferenceDataV1({}, 200));
    render(
      <ChangeOfNameView
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        optional={{}}
      />,
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.checkSupportingDocumentsAppointment),
    ).resolves.toBeInTheDocument();
    await userEvent.click(
      screen.getByLabelText(
        locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
      ),
    );
    await expect(screen.findByText("Date")).resolves.toBeInTheDocument();
    await expect(screen.findByText("Time")).resolves.toBeInTheDocument();
    await typeDateTime(screen, userEvent, "2099");
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeEnabled();
    await user.click(bookAppointmentButton);
    await expect(
      screen.findByText("There was a problem with completing the action"),
    ).resolves.toBeInTheDocument();
  });

  test("it renders requested via des and book appointment", async () => {
    render(
      <ChangeOfNameView
        process={mockDocumentsRequestedDesAppointment}
        mutate={() => {}}
        optional={{}}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.viewInDes),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
  });

  test("it displays change button if date is future", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2099-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
    );

    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(locale.change)).resolves.toBeInTheDocument();
  });

  test("it displays reschedule button if date is past", async () => {
    server.use(getReferenceDataV1({}, 200));
    render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2010-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
    );
    await expect(screen.queryByText(locale.bookAppointment)).toBeNull();
    const button = screen.getByText(locale.reschedule);
    fireEvent.click(button);
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeDisabled();
  });

  test("it enables next once all checkboxes are selected", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
    );

    const nextButton = await screen.findByText(locale.next);
    await expect(nextButton).toBeDisabled();
    await selectAllCheckBoxes();
    await expect(nextButton).toBeEnabled();
  });

  test("it displays an error if there's an issue with confirm", async () => {
    server.use(patchProcessV1("error", 500));
    render(
      <ChangeOfNameView
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
        optional={{}}
      />,
    );
    await expect(screen.findByText(locale.next)).resolves.toBeInTheDocument();
    const nextButton = screen.getByText(locale.next);
    await selectAllCheckBoxes();
    expect(nextButton).toBeEnabled();
    await userEvent.click(nextButton);
    await expect(
      screen.findByText("There was a problem with completing the action"),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewDocuments correctly on state DocumentsRequestedAppointment", async () => {
    jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
    const { container } = render(
      <ReviewDocumentsView
        processConfig={processes.changeofname}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2099-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
        setGlobalError={() => {}}
      />,
      options,
    );
    await expect(screen.queryByText(locale.views.reviewDocuments.viewInDes)).toBeNull();
    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(/08:59 am/)).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(/Monday 12th October 2099/),
    ).resolves.toBeInTheDocument();
    await expect(screen.getByText(locale.next)).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it renders ReviewDocuments correctly on state=ProcessClosed and previousState=DocumentsRequestedDes", async () => {
    jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
    const { container } = render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ProcessCancelled" },
          previousStates: [
            { ...mockProcessV1.currentState, state: "DocumentsRequestedDes" },
          ],
        }}
        mutate={() => {}}
        optional={{
          person: mockPersonV1,
          closeProcessReason: "Test",
        }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.viewInDes),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders ReviewDocuments correctly on state=ProcessClosed and previousState=DocumentsRequestedAppointment", async () => {
    jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
    const { container } = render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ProcessCancelled" },
          previousStates: [
            {
              ...mockProcessV1.currentState,
              state: "DocumentsRequestedAppointment",
              processData: {
                formData: {
                  appointmentDateTime: "2099-10-12T08:59:00.000Z",
                },
                documents: [],
              },
            },
          ],
        }}
        mutate={() => {}}
        optional={{
          person: mockPersonV1,
          closeProcessReason: "Test",
        }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

async function selectAllCheckBoxes() {
  await userEvent.click(
    screen.getByLabelText(locale.views.reviewDocuments.seenPhotographicId),
  );
  await userEvent.click(screen.getByLabelText(locale.views.reviewDocuments.seenSecondId));
  await userEvent.click(
    screen.getByLabelText(locale.views.reviewDocuments.atLeastOneDocument, {
      exact: false,
    }),
  );
}
