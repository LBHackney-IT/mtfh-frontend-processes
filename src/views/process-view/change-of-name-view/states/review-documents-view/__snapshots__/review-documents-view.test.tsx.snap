// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`review-documents-view it renders ReviewDocuments correctly on state DocumentsRequestedAppointment 1`] = `
<div>
  <div
    data-testid="changeofname-ReviewDocuments"
  >
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment scheduled
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
          <div
            style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 15px;"
          >
            <button
              class="govuk-link lbh-link"
              type="button"
            >
              Change
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      style="padding-bottom: 35px;"
    />
    <form
      action="#"
      class="review-documents-form"
      id="review-documents-form"
      novalidate=""
    >
      <h4
        class="lbh-heading-h4"
      >
        Use the form below to record the documents you have checked:
      </h4>
      <div
        class="govuk-form-group lbh-form-group"
        id="review-documents-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="review-documents-form-group-field"
        >
          
          
        </label>
        <div
          class="govuk-checkboxes lbh-checkboxes"
          id="review-documents-form-group-field"
        >
          <div
            class="govuk-checkboxes__item"
          >
            <input
              aria-describedby="seen-photographic-id-hint"
              class="govuk-checkboxes__input"
              id="seen-photographic-id"
              name="seenPhotographicId"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="seen-photographic-id"
            >
              I confirm I have seen a government issue photographic ID
            </label>
            <span
              class="govuk-hint govuk-checkboxes__hint lbh-hint"
              id="seen-photographic-id-hint"
            >
              (for example: valid passport, driving licence)
            </span>
          </div>
          <div
            class="govuk-checkboxes__item"
          >
            <input
              aria-describedby="seen-second-id-hint"
              class="govuk-checkboxes__input"
              id="seen-second-id"
              name="seenSecondId"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="seen-second-id"
            >
              I confirm I have seen a second form of ID (does not have to be photographic)
            </label>
            <span
              class="govuk-hint govuk-checkboxes__hint lbh-hint"
              id="seen-second-id-hint"
            >
              (for example: utility bill, bank statement, council letter)
            </span>
          </div>
          <div
            class="govuk-checkboxes__item"
          >
            <input
              class="govuk-checkboxes__input"
              id="confirmation-for-valid-documents"
              name="atLeastOneDocument"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="confirmation-for-valid-documents"
            >
              I confirm I have seen a valid example of one of the following documents
              <br />
              1
              .
              Marriage certificate
              <br />
              2
              .
              Civil partnership certificate
              <br />
              3
              .
              Decree absolute
              <br />
              4
              .
              Final order
              <br />
              5
              .
              Deed poll document
              <br />
              6
              .
              Statutory declaration
            </label>
          </div>
        </div>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 222px;"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
</div>
`;

exports[`review-documents-view it renders ReviewDocuments correctly on state DocumentsRequestedDes 1`] = `
<div>
  <h2
    class="lbh-heading-h2"
  >
    <div>
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
        target="_blank"
      >
        Haylee
         
        Macejkovic
      </a>
    </div>
  </h2>
  <div
    class="mtfh-box mtfh-box--success"
  >
    <div
      class="mtfh-status-heading"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#00664F"
          stroke-width="1.5"
        />
        <path
          clip-rule="evenodd"
          d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
          fill="#00664F"
          fill-rule="evenodd"
        />
      </svg>
      <h4
        class="lbh-heading-h4 mtfh-status-heading__title"
      >
        Documents requested
      </h4>
    </div>
    <div
      class="govuk-link lbh-link lbh-link--no-visited-state"
      style="margin-left: 60px; margin-top: 17.5px;"
    >
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="http://des/teams/1/dashboard/requests"
        target="_blank"
      >
        View request in Document Evidence Store
      </a>
    </div>
  </div>
  <div
    data-testid="changeofname-ReviewDocuments"
  >
    <div
      class="govuk-checkboxes__item"
    >
      <input
        class="govuk-checkboxes__input"
        id="condition"
        type="checkbox"
      />
      <label
        class="govuk-label govuk-checkboxes__label"
        for="condition"
      >
        I need to make an appointment to check supporting documents
      </label>
    </div>
    <div
      style="padding-bottom: 35px;"
    />
    <form
      action="#"
      class="review-documents-form"
      id="review-documents-form"
      novalidate=""
    >
      <h4
        class="lbh-heading-h4"
      >
        Use the form below to record the documents you have checked:
      </h4>
      <div
        class="govuk-form-group lbh-form-group"
        id="review-documents-form-group"
      >
        <label
          class="govuk-label lbh-label"
          for="review-documents-form-group-field"
        >
          
          
        </label>
        <div
          class="govuk-checkboxes lbh-checkboxes"
          id="review-documents-form-group-field"
        >
          <div
            class="govuk-checkboxes__item"
          >
            <input
              aria-describedby="seen-photographic-id-hint"
              class="govuk-checkboxes__input"
              id="seen-photographic-id"
              name="seenPhotographicId"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="seen-photographic-id"
            >
              I confirm I have seen a government issue photographic ID
            </label>
            <span
              class="govuk-hint govuk-checkboxes__hint lbh-hint"
              id="seen-photographic-id-hint"
            >
              (for example: valid passport, driving licence)
            </span>
          </div>
          <div
            class="govuk-checkboxes__item"
          >
            <input
              aria-describedby="seen-second-id-hint"
              class="govuk-checkboxes__input"
              id="seen-second-id"
              name="seenSecondId"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="seen-second-id"
            >
              I confirm I have seen a second form of ID (does not have to be photographic)
            </label>
            <span
              class="govuk-hint govuk-checkboxes__hint lbh-hint"
              id="seen-second-id-hint"
            >
              (for example: utility bill, bank statement, council letter)
            </span>
          </div>
          <div
            class="govuk-checkboxes__item"
          >
            <input
              class="govuk-checkboxes__input"
              id="confirmation-for-valid-documents"
              name="atLeastOneDocument"
              type="checkbox"
              value="false"
            />
            <label
              class="govuk-label govuk-checkboxes__label"
              for="confirmation-for-valid-documents"
            >
              I confirm I have seen a valid example of one of the following documents
              <br />
              1
              .
              Marriage certificate
              <br />
              2
              .
              Civil partnership certificate
              <br />
              3
              .
              Decree absolute
              <br />
              4
              .
              Final order
              <br />
              5
              .
              Deed poll document
              <br />
              6
              .
              Statutory declaration
            </label>
          </div>
        </div>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 222px;"
        type="submit"
      >
        Next
      </button>
    </form>
  </div>
  <div
    data-testid="close-case-button"
  >
    <p
      class="lbh-body-m"
    >
      If the documents are not suitable and all avenues to obtain the right documents have been exhausted, then close the case.
    </p>
    <button
      class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
      style="width: 222px;"
      type="button"
    >
      Close Case
    </button>
  </div>
</div>
`;

exports[`review-documents-view it renders ReviewDocuments correctly on state=ProcessClosed and previousState=DocumentsRequestedAppointment 1`] = `
<div>
  <h2
    class="lbh-heading-h2"
  >
    <div>
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
        target="_blank"
      >
        Haylee
         
        Macejkovic
      </a>
    </div>
  </h2>
  <div
    data-testid="changeofname-ReviewDocuments"
  >
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-box"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <div>
          <div
            class="mtfh-status-heading__title"
          >
            <h4
              class="lbh-heading-h4"
            >
              Office appointment missed
            </h4>
          </div>
          <p
            class="lbh-body-m"
            style="margin-top: 15px;"
          >
            Date: 
            Monday 12th October 2099
            <br />
            Time: 
            08:59 am
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mtfh-box mtfh-box--warning"
  >
    <div
      class="mtfh-status-box"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#BE3A34"
          stroke-width="1.5"
        />
        <path
          d="M24.2969 26.6587H20.6035L19.8315 11.8701H25.0688L24.2969 26.6587ZM19.771 31.8354C19.771 30.9878 19.998 30.347 20.4521 29.9131C20.9062 29.4792 21.5672 29.2622 22.4351 29.2622C23.2726 29.2622 23.9185 29.4842 24.3726 29.9282C24.8368 30.3722 25.0688 31.008 25.0688 31.8354C25.0688 32.6326 24.8368 33.2633 24.3726 33.7275C23.9084 34.1816 23.2625 34.4087 22.4351 34.4087C21.5874 34.4087 20.9315 34.1867 20.4673 33.7427C20.0031 33.2886 19.771 32.6528 19.771 31.8354Z"
          fill="#BE3A34"
        />
      </svg>
      <div>
        <div
          class="mtfh-status-heading__title"
        >
          <h4
            class="lbh-heading-h4"
          >
            Change of name application cancelled 
          </h4>
        </div>
        <p
          class="lbh-body-m"
          style="margin-top: 15px;"
        >
          Test
        </p>
      </div>
    </div>
  </div>
  <h3
    class="lbh-heading-h3"
  >
    Thank you for your confirmation
  </h3>
  <ul
    class="lbh-list lbh-list--bullet"
    style="margin-left: 1em;"
  >
    <li>
      <p
        class="lbh-body-s"
      >
        This case is now closed and we have recorded this on the system - that you have sent an outcome letter to the resident. The outcome can be viewed in the activity history.
      </p>
    </li>
  </ul>
  <div
    style="margin-top: 1em;"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/"
    >
      Return to home page
    </a>
  </div>
</div>
`;

exports[`review-documents-view it renders ReviewDocuments correctly on state=ProcessClosed and previousState=DocumentsRequestedDes 1`] = `
<div>
  <h2
    class="lbh-heading-h2"
  >
    <div>
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
        target="_blank"
      >
        Haylee
         
        Macejkovic
      </a>
    </div>
  </h2>
  <div
    class="mtfh-box mtfh-box--success"
  >
    <div
      class="mtfh-status-heading"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#00664F"
          stroke-width="1.5"
        />
        <path
          clip-rule="evenodd"
          d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
          fill="#00664F"
          fill-rule="evenodd"
        />
      </svg>
      <h4
        class="lbh-heading-h4 mtfh-status-heading__title"
      >
        Documents requested
      </h4>
    </div>
    <div
      class="govuk-link lbh-link lbh-link--no-visited-state"
      style="margin-left: 60px; margin-top: 17.5px;"
    >
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="http://des/teams/1/dashboard/requests"
        target="_blank"
      >
        View request in Document Evidence Store
      </a>
    </div>
  </div>
  <div
    data-testid="changeofname-ReviewDocuments"
  />
  <div
    class="mtfh-box mtfh-box--warning"
  >
    <div
      class="mtfh-status-box"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#BE3A34"
          stroke-width="1.5"
        />
        <path
          d="M24.2969 26.6587H20.6035L19.8315 11.8701H25.0688L24.2969 26.6587ZM19.771 31.8354C19.771 30.9878 19.998 30.347 20.4521 29.9131C20.9062 29.4792 21.5672 29.2622 22.4351 29.2622C23.2726 29.2622 23.9185 29.4842 24.3726 29.9282C24.8368 30.3722 25.0688 31.008 25.0688 31.8354C25.0688 32.6326 24.8368 33.2633 24.3726 33.7275C23.9084 34.1816 23.2625 34.4087 22.4351 34.4087C21.5874 34.4087 20.9315 34.1867 20.4673 33.7427C20.0031 33.2886 19.771 32.6528 19.771 31.8354Z"
          fill="#BE3A34"
        />
      </svg>
      <div>
        <div
          class="mtfh-status-heading__title"
        >
          <h4
            class="lbh-heading-h4"
          >
            Change of name application cancelled 
          </h4>
        </div>
        <p
          class="lbh-body-m"
          style="margin-top: 15px;"
        >
          Test
        </p>
      </div>
    </div>
  </div>
  <h3
    class="lbh-heading-h3"
  >
    Thank you for your confirmation
  </h3>
  <ul
    class="lbh-list lbh-list--bullet"
    style="margin-left: 1em;"
  >
    <li>
      <p
        class="lbh-body-s"
      >
        This case is now closed and we have recorded this on the system - that you have sent an outcome letter to the resident. The outcome can be viewed in the activity history.
      </p>
    </li>
  </ul>
  <div
    style="margin-top: 1em;"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/"
    >
      Return to home page
    </a>
  </div>
</div>
`;
